pkgs/by-name/ze/zellij/package.nix

```diff
@@ -0,0 +1,99 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  stdenv,
+  installShellFiles,
+  pkg-config,
+  libiconv,
+  curl,
+  openssl,
+  DiskArbitration,
+  Foundation,
+  mandown,
+  zellij,
+  testers,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "zellij";
+  version = "0.41.1";
+
+  src = fetchFromGitHub {
+    owner = "zellij-org";
+    repo = "zellij";
+    rev = "v${version}";
+    hash = "sha256-EUoJHM0Jm0uFKFeHhtzon/ZRC615SHfYa1gr4RnCNBw=";
+  };
+
+  cargoHash = "sha256-rI3pa0dvC/OVJz8gzD1bM0Q+8OWwvGj+jGDEMSbSb2I=";
+
+  env.OPENSSL_NO_VENDOR = 1;
+
+  # Workaround for https://github.com/zellij-org/zellij/issues/3720
+  postPatch = ''
+    substituteInPlace zellij-utils/Cargo.toml \
+      --replace-fail 'isahc = "1.7.2"' 'isahc = { version = "1.7.2", default-features = false, features = ["http2", "text-decoding"] }'
+  '';
+
+  nativeBuildInputs = [
+    mandown
+    installShellFiles
+    pkg-config
+    (lib.getDev curl)
```

Comment:
I think we can go with this for now, but I think cross-compilation might be broken.

1. curl-sys uses https://docs.rs/pkg-config/latest/pkg_config/#cross-compilation, which seems to require setting certain environment variables for cross-compilation use case.
2. we need it to use at build-time the .pc files from curl used at runtime, so how do we do that?

I don't have enough experience in cross-compilation to know how to do this. Maybe @Artturin or someone else can help out here (perhaps as a follow up to this PR)?

Note: This is technically not needed on Darwin since pkg-config is not used on Darwin.

Reply:
Would you make its inclusion conditional here, or do we consider that it doesn't matter to include this path in the sandbox also on darwin and it will simply not be used then?

For the cross-compilation, I also don't have much experience with this.

Reply:
I've been thinking about it a little more, and I think we should just keep it like this for now. There's probably a nice solution for this somewhere, but I'll mull it over some more separately.